Core/MIPS/x86/X64IRAsm.cpp

```diff
@@ -192,21 +210,26 @@ void X64JitBackend::GenerateFixedCode(MIPSState *mipsState) {
 #elif PPSSPP_ARCH(AMD64)
 			MOV(32, R(SCRATCH1), MComplex(MEMBASEREG, SCRATCH1, SCALE_1, 0));
 #endif
-			MOV(32, R(EDX), R(SCRATCH1));
 			_assert_msg_(MIPS_JITBLOCK_MASK == 0xFF000000, "Hardcoded assumption of emuhack mask");
-			SHR(32, R(EDX), Imm8(24));
-			CMP(32, R(EDX), Imm8(MIPS_EMUHACK_OPCODE >> 24));
+			if (cpu_info.bBMI2) {
+				RORX(32, EDX, R(SCRATCH1), 24);
+				CMP(8, R(EDX), Imm8(MIPS_EMUHACK_OPCODE >> 24));
+			} else {
+				MOV(32, R(EDX), R(SCRATCH1));
+				SHR(32, R(EDX), Imm8(24));
+				CMP(32, R(EDX), Imm8(MIPS_EMUHACK_OPCODE >> 24));
+			}
 			FixupBranch needsCompile = J_CC(CC_NE);
 				// Mask by 0x00FFFFFF and extract the block jit offset.
 				AND(32, R(SCRATCH1), Imm32(MIPS_EMUHACK_VALUE_MASK));
 #if PPSSPP_ARCH(X86)
-				ADD(32, R(SCRATCH1), ImmPtr(GetBasePtr()));
+				LEA(32, SCRATCH1, MDisp(SCRATCH1, (u32)GetBasePtr()));
 #elif PPSSPP_ARCH(AMD64)
-				if (jo.reserveR15ForAsm) {
+				if (jitbaseInR15) {
 					ADD(64, R(SCRATCH1), R(JITBASEREG));
 				} else {
 					// See above, reserveR15ForAsm is used when above 0x7FFFFFFF.
-					ADD(64, R(SCRATCH1), Imm32((u32)jitbase));
+					LEA(64, SCRATCH1, MDisp(SCRATCH1, (u32)jitbase));
```

Comment:
is this lea really beneficial? it has the same source and dest register so equivalent to an add, no? Although I guess it's probably entirely neutral since this LEA can be dispatched to any execution unit, and they're the same size, right?

Reply:
Yeah, it did seem a bit neutral in performance but I figured, might as well.  I was trying to do things to reduce time spent in fixed code a bit...

-[Unknown]